nym-wallet/src-tauri/src/state.rs

```diff
@@ -437,14 +437,49 @@ impl WalletStateInner {
         }
     }
 
-    pub fn select_nyxd_url(&mut self, url: &str, network: Network) -> Result<(), BackendError> {
+    pub async fn select_nyxd_url(
+        &mut self,
+        url: &str,
+        network: Network,
+    ) -> Result<(), BackendError> {
+        if !nym_validator_client::connection_tester::test_nyxd_url_connection(
```

Comment:
Pondering if this connection test maybe should be moved to the parent function? (The tauri function)